lib/random_word.rb

```diff
@@ -9,22 +9,35 @@
 module RandomWord
   module EachRandomly
     attr_accessor :random_word_exclude_list
+    attr_accessor :min_length
+    attr_accessor :max_length
 
     def each_randomly(&blk)
       used = Set.new
+      skipped = Set.new
       exclude_list = Array(@random_word_exclude_list)
       while true
         idx = next_unused_idx(used)
         used << idx
         word = at(idx)
-        next if exclude_list.any?{|r| r === word }
+        if exclude_list.any? { |r| r === word } || word.length < min_length || (!(max_length.nil?) && word.length > max_length)
+          skipped << idx
+          next
+        end
         yield word
+        used.subtract(skipped)
+        skipped.clear
       end
 
     rescue OutOfWords
       # we are done.
     end
 
+    def set_constraints(opts)
+      @min_length = opts[:not_shorter_than] || 1
+      @max_length = opts[:not_longer_than] || nil
```

Comment:
I prefer doing something like this, but didn't know this in particular was an option. I was thinking too much about integers, in which case I don't like using `MAX_INT` or whatever that is in Ruby. Do you consider this to be idiomatic Ruby? It makes me feel a little funny, but I could get used to it.